tasks/kubelet_args.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: config | component | kubelet args | check bkp
+  become: true
+  stat:
+    path: /var/snap/microk8s/current/args/kubelet.orig
+  register: kubelet_orig_file
+
+- name: config | component | kubelet args | create bkp
+  become: true
+  ansible.builtin.copy:
+    src: /var/snap/microk8s/current/args/kubelet
+    dest: /var/snap/microk8s/current/args/kubelet.orig
+    remote_src: yes
+  when: not kubelet_orig_file.stat.exists
+  notify: restart kubelite
+
+- name: config | component | kubelet args | restore orig config
+  become: true
+  ansible.builtin.copy:
+    src: /var/snap/microk8s/current/args/kubelet.orig
+    dest: /var/snap/microk8s/current/args/kubelet
+    remote_src: yes
+  when: kubelet_orig_file.stat.exists
+  notify: restart kubelite
+
+- name: config | component | kubelet args | append args
+  become: true
+  loop: "{{ microk8s_kubelet_additional_args }}"
+  lineinfile:
+    path: /var/snap/microk8s/current/args/kubelet
+    line: "{{ item }}"
+    state: present
+  notify: restart kubelite
```

Comment:
This is not idempotent though. To make it idempotent:
1. `slurp` the content of the orig.bkp into a variable (instead of `config | component | kubelet args | restore orig config`)
2. use a `template/kubelete_args.j2`, where that variable will be dumped first and kubelet args will be iterated over (instead of `config | component | kubelet args | append args`)

That way the change will only happen, if the template result does not match what is currently present in the target kubelet args config.

Reply:
The template contents:

```jinja2
{{ _kubelet_args_orig_bkp }}
{% for item in data if microk8s_kubelet_additional_args %}
{{ item}}
{% endfor %}
```